### PR TITLE
Configure IaD manifests

### PR DESCRIPTION
**Pull request by vedantthapa**
As per the recent internal discussions, `acm-core` only hosts infrastructure upto the project-level in GCP. We were, therefore, asked to move any infrastructure components that go beyond the project-level to the application repo.

This, however, creates a bootstrapping issue since a k8s cluster is required before any infrastructure manifests can be applied. To resolve this, we use a [Taskfile](https://taskfile.dev/) at `k8s/Taskfile.yaml` to keep track of any resources provisioned to get the k8s cluster up and running in the new GCP project space. Once the k8s cluster is ready, Flux and [KCC](https://cloud.google.com/config-connector/docs/overview) will provision the rest of the setup, which, is being tracked in the `infrastructure/` directory at the root of this repo.

Summary of resources provisioned outside the k8s cluster:
- Enable container and mesh apis
- VPC and subnet in `northamerica-northeast1` region
- IAM service account with `Owner` permissions for KCC to provision other resources.
- Role-bindings for cross-project fleet setup and anthos.
- Provision anthos service mesh

Other than these, all other resources will / should be added in the `infrastructure/` dir as KCC manifests.

Relates to https://github.com/PHACDataHub/cpho-phase2/pull/153 and #135

### CODE REVIEW

**Review by Stephen-ONeil — CHANGES_REQUESTED** [review]
So we're already off kpt? Are we going to move back to kustomize for templating, variables, etc?

File: infrastructure/bucket.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    cnrm.cloud.google.com/force-destroy: "false"
+  labels:
+    vanity-name: "pht-hopic" 
+    classification: ucll 
+spec:
+  location: northamerica-northeast1
+  versioning:
+    enabled: true
+  storageClass: STANDARD
+  uniformBucketLevelAccess: true
```
[the remark this answers is not in the document]
Comment (vedantthapa) [reply infrastructure/bucket.yaml:35]: added with [ed2ffd3](https://github.com/PHACDataHub/cpho-phase2/pull/193/commits/ed2ffd34eb30232e02a2ba02e6d43a0e5f5a65bd)

**Comment by vedantthapa** [comment]
@Stephen-ONeil 
> So we're already off kpt? Are we going to move back to kustomize for templating, variables, etc?

Yep! 

Good catch, almost forgot I had to add a `kustomization.yaml`. I've also moved all common labels to the kustomization file.

**Review by Stephen-ONeil — CHANGES_REQUESTED** [review]
`CloudBuildTrigger` should be good to enable now, let's do that and confirm it's functioning as expected. Think we're good to merge after that :+1:

**Pull request merged**